### ISSUE TITLE
Allow for overriding of base locale

### ISSUE THREAD

**Issue by mikeakers** [mac-cain13/R.swift#539]
In our project the comments in our generated localized strings struct contains the catalan (ca) translations of our strings instead of the "en" version which is what we would expect.

example:

```Swift
/// ca translation: Cancel·la
/// 
/// Locales: ca, de, en, es-419, es-ES, fr, fr-CA, id, ja, ko, ms, nl, pl, pt-BR, ru, th-TH, vi, zh-CN, zh-TW
static let genericCancel = Rswift.StringResource(key: "generic.cancel", tableName: "Localizable", bundle: R.hostingBundle, locales: ["ca", "de", "en", "es-419", "es-ES", "fr", "fr-CA", "id", "ja", "ko", "ms", "nl", "pl", "pt-BR", "ru", "th-TH", "vi", "zh-CN", "zh-TW"], comment: nil)
```

Reading through `StringsStructGenerator` and `Locale` it looks like R.swift is looking for an `.lproj` named `Base` to take values from, and if that fails it fall back to the first locale, which explains why we see values for `ca` in the comments.

Due to integrations with 3rd party translations services changing our .lproj names would be difficult. Is there a way I'm missing to override the "Base" locale? Ideally we'd like a way to set it to `en`.

If R.swift doesn't support setting the base locale, what do you think of an enhancement that would allow passing in the base locale as a parameter or env variable?

**Comment by scalbatty** [comment]
Maybe if there is no `Base` version, the fallback could be the development language extracted from inside the `pbxproj` file at:

```
/* Begin PBXProject section */
		/* Project object */ = {
			isa = PBXProject;
			developmentRegion = en;
```

That would be a natural solution that doesn't require any manual configuration.

**Comment by tomlokhorst** [comment]
Yes, that's my thinking as well. To use the developmentRegion as the default instead of Base.
I think as of Xcode 10?, Base isn't used anymore and there's always a developmentRegion set to a specific language.

**Comment by scalbatty** [comment]
It seems to be the case indeed.

@tomlokhorst I'm pretty new to the project, what do you think the best approach is?

- add a `isDevelopmentRegion` flag to `LocalizableString` or `Locale` when parsing the resources?
- inject the `developmentRegion` language code into the `StringsStructGenerator` and add a rule to the `comments` computed variable?

**Comment by tomlokhorst** [comment]
Yes something like that. I'm currently working on implementing this.

**Comment by tomlokhorst** [comment]
I've created an the implementation (https://github.com/mac-cain13/R.swift/pull/550) and tested it with some older projects. Those still have a Base translation, and no translations matching the developmentRegion.

The way I've implemented it is to first use the Base locale, if that isn't available, use the developmentRegion from the Xcode project. If that translation also doesn't exist, it will fallback to the first of the translations from the Xcode project.
